src/test/java/de/tum/in/www1/artemis/StudentExamIntegrationTest.java

```diff
@@ -70,7 +70,13 @@ public void initTestCase() {
     }
 
     @AfterEach
-    public void resetDatabase() {
+    public void resetDatabase() throws Exception {
+
+        // change back to instructor user
+        database.changeUser("instructor1");
+        // Clean up to prevent exceptions during reset database
+        request.delete("/api/courses/" + course1.getId() + "/exams/" + exam1.getId(), HttpStatus.OK);
```

Comment:
We should find a better solution for this soon :-D

Reply:
In my project I always use a try{}finally{} and within the finally I delete all of the entities created by accessing the repository directly
if(repo.existsById(...)){
    repo.deleteById(...);
}
Something like that. Is that something that might help you?

Reply:
we have a better solution already, I added a cleanup database service on `develop` so this line might not even be necessary, but still a good test that the DELETE Rest call does work as intended